{"description": "In versions prior 0.11.3 it's possible to make from_slice panic by feeding it certain malformed input. It's never documented that from_slice (and from_bytes which wraps it) can panic, and its' return type (Result<Self, DecodeError>) suggests otherwise. In practice, from_slice/from_bytes is frequently used in networking code and is being called with unsanitized data from untrusted sources. This can allow attackers to cause DoS by causing an unexpected panic in the network client's code..", "methods": ["Feeds unsanitized data from untrusted sources to from_slice/from_bytes.", "Causes an unexpected panic in the network client's code.", "Allows for a Denial of Service (DoS) attack."]}